### PR TITLE
Enable GraphQL master tests

### PR DESCRIPTION
**Pull request by lrafeei**
This PR 
1. enables the testing of graphqlmaster.  That is, it pulls graphql-core's code from the main branch as well as the latest release to make sure future releases of graphql-core do not conflict with New Relic's instrumentation/crash customer code
2. adds/tweaks instrumentation to support graphql-core v3.3.  Currently, this is still an alpha version
3. updates GraphQL tests to support both v3.2 and v3.3+

### CODE REVIEW

File: tests/framework_graphql/test_application.py [hunk 1]
```diff
@@ -172,8 +177,17 @@ def test_query_and_mutation(target_application):
...
+        validate_code_level_metrics(
+            "framework_%s.test_application.test_query_and_mutation.<locals>" % (framework.lower()), "_mutation"
+        ),
+        VERSION_3_3,
+    )
+    @conditional_decorator(
+        validate_code_level_metrics(
+            "framework_%s._target_schema_%s" % (framework.lower(), schema_type), "resolve_storage_add"
+        ),
+        not VERSION_3_3,
```
Comment (TimPansino) [inline tests/framework_graphql/test_application.py:190]: This seems like a fault in CLM

File: tests/framework_graphql/test_application.py [hunk 2]
```diff
@@ -193,7 +207,18 @@ def _mutation():
...
+            "framework_%s.test_application.test_query_and_mutation.<locals>" % (framework.lower()), "_query"
+        ),
+        VERSION_3_3,
+    )
+    @conditional_decorator(
+        validate_code_level_metrics(
+            "framework_%s._target_schema_%s" % (framework.lower(), schema_type), "resolve_storage"
+        ),
+        not VERSION_3_3,
+    )
```
Comment (TimPansino) [inline tests/framework_graphql/test_application.py:221]: This does as well

File: tests/framework_graphql/test_application.py [hunk 3]
```diff
@@ -309,7 +334,11 @@ def test_exception_in_resolver(target_application, field):
     framework, version, target_application, is_bg, schema_type, extra_spans = target_application
     query = "query MyQuery { %s }" % field
 
-    txn_name = "framework_%s._target_schema_%s:resolve_error" % (framework.lower(), schema_type)
+    txn_name = (
+        "graphql.execution.execute:ExecutionContext.execute_operation"
+        if VERSION_3_3
+        else "framework_%s._target_schema_%s:resolve_error" % (framework.lower(), schema_type)
+    )
```
Comment (TimPansino) [inline tests/framework_graphql/test_application.py:341]: This looks like a fault in transaction naming

File: tests/framework_graphql/test_application.py [hunk 4]
```diff
@@ -414,10 +443,12 @@ def test_operation_metrics_and_attrs(target_application):
         "graphql.operation.name": "MyQuery",
     }
 
-    # Span count 16: Transaction, Operation, and 7 Resolvers and Resolver functions
+    # Span count 16 for v3.2: Transaction, Operation, and 7 Resolvers and Resolver functions
+    # Span count 15 for v3.2: Transaction, Operation, and 6 Resolvers and Resolver functions
```
Comment (TimPansino) [inline tests/framework_graphql/test_application.py:447]: Unless they're optimizing away some resolvers, I think we have an issue with span count as well

File: tests/framework_graphql/test_application.py [hunk 5]
```diff
@@ -147,9 +151,10 @@ def test_query_and_mutation(target_application):
         ("GraphQL/operation/%s/mutation/<anonymous>/%s" % (framework, mutation_path), 1),
     ]
     _test_query_scoped_metrics = [
-        ("GraphQL/resolve/%s/storage" % framework, 1),
         ("GraphQL/operation/%s/query/<anonymous>/storage" % framework, 1),
     ]
+    if not VERSION_3_3:
+        _test_query_scoped_metrics.append(("GraphQL/resolve/%s/storage" % framework, 1))
```
Comment (TimPansino) [inline tests/framework_graphql/test_application.py:157]: What happened here? Is this resolver trace somehow gone?

Comment (lrafeei) [reply tests/framework_graphql/test_application.py]: Whoops.  That's residual from the previous nonsense.  Fixed.

**Review by TimPansino — APPROVED** [review]

**Pull request merged**